packages/pigeon/tool/shared/generation.dart

```diff
@@ -48,60 +48,79 @@ String _javaFilenameForName(String inputName) {
 }
 
 Future<int> generateExamplePigeons() async {
-  return runPigeon(
+  int success = 0;
+  success = await runPigeon(
     input: './example/app/pigeons/messages.dart',
     basePath: './example/app',
     suppressVersion: true,
   );
+  success = await runPigeon(
```

Comment:
You haven't checked the previous result; if the call above failed, this will continue, and then throw away that failure information.